File: Paradigma_Orientado_a_Objetos/C#/StatisticsDisplay.cs
using System;
using System.Text;

namespace HeadFirstDesignPatterns.Observer.WeatherData
{
	public class StatisticsDisplay : IObserver, IDisplayElement
	{
		private float maxTemp = 0.0f;
		private float minTemp = 200;//set intial high so that minimum 
									//is set first invokation
		private float temperatureSum = 0.0f;
		private int numReadings = 0;
		private ISubject weatherData;
		
		public int NumberOfReadings
		{
			get
			{
				return numReadings;
			}
		}
		
		public StatisticsDisplay(ISubject weatherData)
		{
			this.weatherData = weatherData;
			weatherData.RegisterObserver(this);
		}

		public void Update(float temperature, float humidity, float pressure)
		{
			temperatureSum += temperature;
			numReadings++;

			if (temperature > maxTemp) 
			{
				maxTemp = temperature;
			}
 
			if (temperature < minTemp) 
			{
				minTemp = temperature;
			}
		}

		public object Display()
		{
			return "Avg/Max/Min temperature = " + RoundFloatToString(temperatureSum / numReadings) + 
				"F/" + maxTemp + "F/" + minTemp + "F";
		}

		public static string RoundFloatToString(float floatToRound)
		{
			System.Globalization.CultureInfo cultureInfo = new System.Globalization.CultureInfo("en-US");
			cultureInfo.NumberFormat.CurrencyDecimalDigits = 2;
			cultureInfo.NumberFormat.CurrencyDecimalSeparator = ".";
			return floatToRound.ToString("F",cultureInfo);
		}
	}
}

File: Paradigma_Orientado_a_Objetos/Java/src/Observer/weather/StatisticsDisplay.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Observer.weather;

public class StatisticsDisplay implements ObserverInterface, DisplayElement {
  private float maxTemp = 0.0f;
  private float minTemp = 200;
  private float tempSum = 0.0f;
  private int numReadings;
  private Subject weatherData;
  
  public StatisticsDisplay(Subject weatherData) {
    this.weatherData = weatherData;
    weatherData.registerObserver(this);
  }
  
  public void update(float temperature, float humidity, float pressure) {
    tempSum += temperature;
    numReadings++;
    
    if (temperature > maxTemp) {
      maxTemp = temperature;
    }
    
    if (temperature < minTemp) {
      minTemp = temperature;
    }
    
    display();
  }
  
  public void display() {
    System.out.println("Avg/Max/Min temperature = " + (tempSum / numReadings) + "/" + maxTemp + "/" + minTemp);
  }
}